src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+import { getArticleById } from '../utils/api';
+
+const SingleArticle = () => {
+	const [article, setArticle] = useState({});
+	const { article_id } = useParams();
+
+	useEffect(() => {
```

Comment:
At some point we are going to want to hand non-existent id's - we would like to give the client something a bit more constructive than the generic 404 that is being thrown.